OneCommunityGlobal/HighestGoodNetworkApp
# Navneeth add implementation for resolveTask permission

# Description
![task description](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/44264834/918a9452-aa3a-426c-b628-1f9a973dad42)


## Related PRs (if any):
Please test the PR with the latest development branch on the backend.

## Main changes explained:
- Update check for whether an user is able to resolve a task.
- Call a dispatch to check if the user has the `resolveTask` permission.
- Add a toast message on successful 'mark as done' operation on a task.
- Remove unused state variable `shouldFetchData` in `src/components/Timelog/Timelog.jsx

## How to test:
1. Check into the current branch.
2. Make the following code change in `src/components/TeamMemberTasks/TeamMemberTask.jsx` for easier testing:
- Replace line 65: 
```javascript
const rolesAllowedToResolveTasks = ['Administrator', 'Owner'];
```
with: 
```javascript
const rolesAllowedToResolveTasks = ['Owner'];
```
4. Run the `development` branch of the backend with `npm install` and `npm run dev`.
5. Run this frontend branch with `npm install` and `npm run start:local`.
6. Clear site data/cache.
7. Log in as Owner -> Other Links -> Permissions Management -> Manage User Permissions.
8. Enter the username of an Admin user that does not have the “Resolve Tasks” permission. Add the permission and Click on Submit.
9. Login as the chosen Admin user (if already logged in, make sure to log out and log back in). The user should be able to see the ‘green tick’ icon for tasks displayed on Dashboard -> Tasks.
10. Resolve a task successfully by clicking on the green tick and verify if it is no longer present. See the third test video 'Adding a new task' for instructions on creating a new task.

## Test Videos (with instructions narrated):

### Code Changes
https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/44264834/18de107e-a284-45b7-b96e-56947df9e4e4

### Testing
https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/44264834/82faaf12-6379-44e2-8dec-d1562024586d

### Adding a new task

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/44264834/a66b5ed6-614e-41a3-8133-13cba00eb3fc
